static/app/views/issueDetails/streamline/hooks/useCopyIssueDetails.spec.tsx

```diff
@@ -0,0 +1,364 @@
+import {EventFixture} from 'sentry-fixture/event';
+import {GroupFixture} from 'sentry-fixture/group';
+
+import {renderHook} from 'sentry-test/reactTestingLibrary';
+
+import * as indicators from 'sentry/actionCreators/indicator';
+import {
+  type AutofixData,
+  AutofixStatus,
+  AutofixStepType,
+} from 'sentry/components/events/autofix/types';
+import * as autofixHooks from 'sentry/components/events/autofix/useAutofix';
+import type {GroupSummaryData} from 'sentry/components/group/groupSummary';
+import * as groupSummaryHooks from 'sentry/components/group/groupSummary';
+import type {EventTag} from 'sentry/types/event';
+import {useCopyIssueDetails} from 'sentry/views/issueDetails/streamline/hooks/useCopyIssueDetails';
+
+// Mock the internal helper function since it's not exported
+jest.mock('sentry/views/issueDetails/streamline/hooks/useCopyIssueDetails', () => {
+  // Store the original implementation
+  const originalModule = jest.requireActual(
+    'sentry/views/issueDetails/streamline/hooks/useCopyIssueDetails'
+  );
+
+  // Mock the internal function for our tests
+  const mockIssueAndEventToMarkdown = jest.fn(
+    (group, event, groupSummaryData, autofixData) => {
+      let text = `# ${group.title}\n\n`;
+      text += `**Issue ID:** ${group.id}\n`;
+      if (group.project?.slug) {
+        text += `**Project:** ${group.project.slug}\n`;
+      }
+
+      if (groupSummaryData) {
+        text += `## Issue Summary\n${groupSummaryData.headline}\n`;
+        text += `**What's wrong:** ${groupSummaryData.whatsWrong}\n`;
+        if (groupSummaryData.trace) {
+          text += `**In the trace:** ${groupSummaryData.trace}\n`;
+        }
+        if (groupSummaryData.possibleCause && !autofixData) {
+          text += `**Possible cause:** ${groupSummaryData.possibleCause}\n`;
+        }
+      }
+
+      if (autofixData) {
+        text += `\n## Root Cause\n`;
+        text += `\n## Solution\n`;
+      }
+
+      if (event.tags && event.tags.length > 0) {
+        text += `\n## Tags\n\n`;
+        event.tags.forEach((tag: EventTag) => {
+          if (tag && typeof tag.key === 'string') {
+            text += `- **${tag.key}:** ${tag.value}\n`;
+          }
+        });
+      }
+
+      // Add mock exception info so we can test that part as well
+      text += `\n## Exception\n`;
+
+      return text;
+    }
+  );
+
+  // Replace the original implementation with our mock
+  return {
+    ...originalModule,
+    // Export the mock for testing
+    __esModule: true,
+    __mocks__: {
+      issueAndEventToMarkdown: mockIssueAndEventToMarkdown,
+    },
+  };
+});
+
+// Get access to our mocked function
+const issueAndEventToMarkdown = jest.requireMock(
+  'sentry/views/issueDetails/streamline/hooks/useCopyIssueDetails'
+).__mocks__.issueAndEventToMarkdown;
```

Comment:
this should be an import at the top of the document and use jest.mocked(issueAndEventToMarkdown) if you need the types